cmd/der2ascii/writer.go

```diff
@@ -355,6 +355,44 @@ func derToASCIIImpl(out *bytes.Buffer, in []byte, indent int, stopAtEOC bool) []
 					// Emit the remaining as a DER element.
 					derToASCIIImpl(out, elem.body[1:], indent+1, false) // Adds a trailing newline.
 					addLine(out, indent, "}")
+				} else if len(elem.body) == 1 && elem.body[0] == 0 {
+					addLine(out, indent, fmt.Sprintf("%s b`` }", header))
+				} else if len(elem.body) > 1 && len(elem.body) <= 5 && elem.body[0] < 8 {
+					// Convert to a b`` literal when the leading bit is valid and the
+					// number of data octets is at most 4.
+					//
+					// The 4 octet limit is as above: X.509 incorrectly encodes things
```

Comment:
Oh huh. Between the comments, the code, the test, and the spec, we seem a bit undecided about the ordering here. Suppose some input is both made of elements _and_ is a valid, short bit string.

* The spec says bit string literal wins.
* The code says elements wins.
* This comment is written as if bit string literal wins. (If elements wins, the X.509 thing doesn't figure into the length limit.)
* The test is written as if elements win, both in expectations and because the various "BIT STRINGs are encoded normally" tests only make sense if elements win. (If bit strings win, I think we'd want them to use long inputs, to test that the elements check is sensitive to non-zero padding, etc.)

I don't feel strongly about which one. I think both are plausible. (You could argue that a short bit string is probably a bitmask because it's short and anything X.509 sticks in BIT STRING is long. You could also argue that the `isMadeOfElements` heuristic seems to be reasonably good in practice and so we may as well let it win.)

Reply:
I think the comment is just worded poorly and I didn't think about the spec too hard. Elements should always win if possible. I'll update the comment and the spec to make that clear.